azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -758,7 +761,10 @@ void shutdown() {
       logger.error("Error shutting down JettyServer while winding down the FlowContainer", e);
     }
     logger.info("Sayonara!");
-    uploadLogFile(execId);
+    if (this.flowRunner != null) {
```

Comment:
I don't like this. If this flowRunner is null wouldn't the .getExecutionId() call above give an NPE?